<?xml version="1.0" ?>

<!--
To check that you've selected the right clusters, play back the generated bagfile with the "pause" option and step through each frame to see where the red markers end up.
-->

<launch>
  <!-- global params -->
  <param name="min_points_per_cluster" value="3" />
  <param name="cluster_dist_euclid" value="0.13" />

   <!-- first rosbag file with positive examples -->
  <node pkg="leg_tracker" type="extract_positive_training_clusters" name="extract_positive_training_clusters1" output="screen">
    <param name="load_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=0.33/positive_not_extracted/corridor1.bag"/>
    <param name="save_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=0.33/corridor1_extracted.bag"/>
    <param name="scan_topic" value="right_scan"/>
    <param name="laser_frame" value="right_laser"/>
    <param name="tf_data" value="/tf"/>
    <rosparam param="polygon_x"> [0,4,4.1,4.5,5.25,11,10.4,5.54,4.5,4.2,4.2,0]</rosparam>
    <rosparam param="polygon_y"> [1.38,1.5,1.45,1.45,1.67,1.85,-0.45,-0.92,-0.75,-0.75,-0.9,-1.13]</rosparam>  

  </node>       

  <node pkg="leg_tracker" type="extract_positive_training_clusters" name="extract_positive_training_clusters2" output="screen">
    <param name="load_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=0.33/positive_not_extracted/corridor2.bag"/>
    <param name="save_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=0.33/corridor2extracted.bag"/>
    <param name="scan_topic" value="/RAWLASER1"/>
    <param name="tf_data" value="/tf"/>
    <rosparam param="polygon_x"> [0,4,4.1,4.5,5.25,11,10.4,5.54,4.5,4.2,4.2,0]</rosparam>
    <rosparam param="polygon_y"> [1.38,1.5,1.45,1.45,1.67,1.85,-0.45,-0.92,-0.75,-0.75,-0.9,-1.13]</rosparam>  

  </node>    

  
  <!-- This is just an example showing you how you can also specify a bounding box for the positive clusters if you want. I don't think it will run because the bag files aren't there. -->
  <!--
  <node pkg="leg_tracker" type="extract_positive_training_clusters" name="extract_positive_training_clusters1" output="screen">
    <param name="load_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=1.0/positive_not_extracted/positive_example.bag"/>
    <param name="save_bag_file" value="$(find leg_tracker)/rosbags/training_scans_res=1.0/positive_example_extracted.bag"/>
    <param name="scan_topic" value="/lasers/front_scan"/>
    <param name="laser_frame" value="front_laser"/>
    <param name="x_min" value ="0.0"/> 
    <param name="x_max" value ="10.0"/> 
    <param name="y_min" value ="-1.2"/> 
    <param name="y_max" value ="0.5"/> 
  </node>
  -->
  


</launch>

